### PR TITLE
Update editor release 2022-10-19

### PR DESCRIPTION
**Pull request by ziegenberg**
This updates the editor to the release 2022-10-19. For the full release notes, please take a look at:

- https://github.com/opencast/opencast-editor/releases/tag/2022-10-06
- https://github.com/opencast/opencast-editor/releases/tag/2022-10-19

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Comment by lkiesow** [comment]
For me, trying to save something using this versions ends in an error like this:

Screenshot from 2022-10-18 16-22-05

**Comment by lkiesow** [comment]
Merging #4317 somewhat helped. Now, saving doesn't fail every time. But if I try setting a thumbnail, it still fails. Now the error message is “Unable to parse details”. But the actual problem seems to be in the backend:

```
2022-10-19 09:20:48,262 | DEBUG | (EditorRestEndpointBase:114) - Unable to parse editing information ({"segments":[{"start":0,"end":4160,"deleted":false,"selected":false},{"start":4160,"end":4163,"deleted":true,"selected":false}],"tracks":[{"audio_stream":{"available":true,"thumbnail_uri":null,"enabled":true},"video_stream":{"available":true,"thumbnail_uri":null,"enabled":true},"flavor":{"type":"presenter","subtype":"preview"},"uri":"http://localhost:8080/static/mh_default_org/internal/01/6feeba2e-03e7-407b-b29b-975fbbde0080/goat.mp4","id":"177a24e1-54ef-4644-af61-2a51bed688ac","thumbnailUri":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAB4AAAAQ4CAYAAADo08FDAAAgAElEQV...HDdb1cJ6Pc8F+OMjDq/KgEZ8c9QGDmL8y7Puvm3D1kD+P8DUqB6MGWzPxYAAAAASUVORK5CYII=","thumbnailPriority":-1}]})
java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
        at org.opencastproject.editor.EditorServiceImpl.addThumbnailsToArchive(EditorServiceImpl.java:617) ~[?:?]
        at org.opencastproject.editor.EditorServiceImpl.setEditData(EditorServiceImpl.java:1064) ~[?:?]
        at org.opencastproject.editor.api.EditorRestEndpointBase.editVideo(EditorRestEndpointBase.java:110) [!/:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) [!/:3.4.6]
        at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) [!/:3.4.6]
        at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) [!/:3.4.6]
        at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) [!/:3.4.6]
        at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) [!/:3.4.6]
        at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) [!/:3.4.6]
        at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308) [!/:3.4.6]
        at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) [!/:3.4.6]
        at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) [!/:3.4.6]
        at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) [!/:3.4.6]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) [!/:3.4.6]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) [!/:3.4.6]
        at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225) [!/:3.4.6]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304) [!/:3.4.6]
        at org.opencastproject.kernel.rest.RestPublisher$RestServlet.handleRequest(RestPublisher.java:615) [!/:?]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217) [!/:3.4.6]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) [!/:3.1.0]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279) [!/:3.4.6]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.rest.JsonpFilter.doFilter(JsonpFilter.java:125) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.filter.proxy.TransparentProxyFilter.doFilter(TransparentProxyFilter.java:75) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.security.urlsigning.filter.UrlSigningFilter.doFilter(UrlSigningFilter.java:100) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.security.RemoteUserAndOrganizationFilter.doFilter(RemoteUserAndOrganizationFilter.java:260) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.rest.CurrentJobFilter.doFilter(CurrentJobFilter.java:102) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:330) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:118) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:84) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.opencastproject.kernel.security.AsyncTimeoutRedirectFilter.doFilter(AsyncTimeoutRedirectFilter.java:60) [!/:?]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:103) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:113) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:146) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:54) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:45) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:98) [!/:3.1.4.RELEASE]
        at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilter(BasicAuthenticationFilter.java:150) [!/:3.1.7.RELEASE]
        at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
        at org.springframework.security.web.authentication.www.DigestAuthenticationFilter.doFilter(DigestAuthenticationFilter.java:115) [!/:3.1.7.RELEASE]
        at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:102) [!/:3.1.4.RELEASE]
        at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:82) [!/:3.1.4.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter.doFilter(DefaultLoginPageGeneratingFilter.java:91) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:183) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:105) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:342) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:192) [!/:3.1.7.RELEASE]
        at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:160) [!/:3.1.7.RELEASE]
        at org.opencastproject.kernel.security.SecurityFilter.doFilter(SecurityFilter.java:138) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.security.OrganizationFilter.doFilter(OrganizationFilter.java:149) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.filter.https.HttpsFilter.doFilter(HttpsFilter.java:85) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.opencastproject.kernel.rest.CleanSessionsFilter.doFilter(CleanSessionsFilter.java:101) [!/:?]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:201) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:228) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548) [!/:9.4.43.v20210629]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) [!/:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:602) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1434) [!/:9.4.43.v20210629]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:298) [!/:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1349) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) [!/:9.4.43.v20210629]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:82) [!/:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.Server.handle(Server.java:516) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:386) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883) [!/:9.4.43.v20210629]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034) [!/:9.4.43.v20210629]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

Not knowing the code, it seems like the editor backend will try to get tags from old thumbnails, but always assumes that there is already at least one preview image and fail if there is none.

Data sent to backend with formatting:

```json
{
  "segments": [
    {
      "start": 0,
      "end": 4160,
      "deleted": false,
      "selected": false
    },
    {
      "start": 4160,
      "end": 4163,
      "deleted": true,
      "selected": false
    }
  ],
  "tracks": [
    {
      "audio_stream": {
        "available": true,
        "thumbnail_uri": null,
        "enabled": true
      },
      "video_stream": {
        "available": true,
        "thumbnail_uri": null,
        "enabled": true
      },
      "flavor": {
        "type": "presenter",
        "subtype": "preview"
      },
      "uri": "http://localhost:8080/static/mh_default_org/internal/01/6feeba2e-03e7-407b-b29b-975fbbde0080/goat.mp4",
      "id": "177a24e1-54ef-4644-af61-2a51bed688ac",
      "thumbnailUri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAB4AAAAQ4CAYAAADo08FDAAAgAElEQV...HDdb1cJ6Pc8F+OMjDq/KgEZ8c9QGDmL8y7Puvm3D1kD+P8DUqB6MGWzPxYAAAAASUVORK5CYII=",
      "thumbnailPriority": -1
    }
  ]
}
```

**Comment by lkiesow** [comment]
Hopefully found the problem. Building a patch for it right now…

**Comment by lkiesow** [comment]
Pull request #4340 should hopefully fix the problem.

**Comment by mliradelc** [comment]
Have this version the subtitle editor? https://github.com/opencast/opencast-editor/pull/638

**Comment by ziegenberg** [comment]
Looking at the linked release notes at the top of the PR, it says, there's only the Thumbnail Editor included.

**Comment by Arnei** [comment]
I am planning on cutting another editor release that includes the subtitle editor, but thought it would be wise to wait with it until this is approved.

File: etc/ui-config/mh_default_org/editor/editor-settings.toml [hunk 1]
```diff
@@ -59,7 +90,19 @@
...
+# specified in the Opencast configuration file
+# `etc/org.opencastproject.editor.EditorServiceImpl.cfg`. It is useful
+# when working with multiple thumbnails.
+# Simple mode only allows users to edit the "primary" thumbnail, as specified
+# in the Opencast configuration file
+# `etc/org.opencastproject.editor.EditorServiceImpl.cfg`. It is useful
+# when there is only a single thumbnail to worry about and you want hide
+# potential fallbacks from the user. If a primary thumbnail cannot be
+# determined, this falls back to professional mode.
+simpleMode = false
```
Comment (lkiesow) [inline etc/ui-config/mh_default_org/editor/editor-settings.toml:108]: This is actually the default, but we can fix that next time.
This doesn't hurt and it's also not set in the upstream config yet:
- https://github.com/opencast/opencast-editor/pull/837

**Comment by lkiesow** [comment]
But with @Arnei's update, it now includes the subtitle editor as well.
It's not really usable by default, though. It needs some workflow modifications at least.

Anyway, seems to work now. Merging…

**Pull request merged**